### PR TITLE
fix: considering git branch for env vars

### PR DESCRIPTION
**Pull request by wagnertimo**
PR to address issue #166:

Only declaring already existing environment variables if also the git branch name is equal (and present, see `processEnvVariable` function):
```js
existingVariable && existingVariable.gitBranch === gitBranch
```

In that case one can declare different environment variable values for the same target (`Preview`) and key (e.g. `DB_URL`) but for different branches (e.g. `foo` and `bar`).

### CODE REVIEW

File: src/VercelEnvVariabler.test.ts [hunk 1]
```diff
@@ -209,4 +223,119 @@ describe("VercelEnvVariabler", () => {
...
+                value: env6BranchBarValue,
+                target: ["preview"],
+                type: "plain",
+                gitBranch: "bar",
+            })
+        );
+    });
+
+    it("Should determine no changes for ENV_6 for foo branch", async () => {
+        // overwrite process.env of ENV_6 for branch bar (set in beforeEach) to old foo branch values (see envVariableFixtures)
```
Comment (dkershner6) [inline src/VercelEnvVariabler.test.ts:267]: `set in BeforeAll`

File: src/VercelEnvVariabler.ts [hunk 2]
```diff
@@ -238,14 +243,18 @@ export default class VercelEnvVariabler {
         gitBranch: string | undefined;
     }) {
         const existingVariable = Object.values(existingVariables)[0]; // They are all actually the same
+        const gitBranchInfoTextSnippet = gitBranch
+            ? ` on git branch '${gitBranch}'`
+            : "";
```
Comment (dkershner6) [inline src/VercelEnvVariabler.ts:248]: 👍

**Review by dkershner6 — CHANGES_REQUESTED** [review]
This looks good to me, lets be sure to get those comments accurate, then ill merge this in for ya.

File: src/VercelEnvVariabler.test.ts (hunk 1 above)
Comment (wagnertimo) [reply src/VercelEnvVariabler.test.ts]: fixed comment to `... (set in beforeAll)... `

File: src/VercelEnvVariabler.ts (hunk 2 above)
Comment (wagnertimo) [reply src/VercelEnvVariabler.ts]: ... also refactored the git branch logging... you might want to a brief look 👀

**Comment by wagnertimo** [comment]
> This looks good to me, lets be sure to get those comments accurate, then ill merge this in for ya.

Thanks for being responsive 🙌

**Pull request merged**